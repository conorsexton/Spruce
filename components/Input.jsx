import React from 'react';
import { render } from 'react-dom';

const Input = props => {
  return (
    <textarea cols={props.cols} value={props.value} onChange={props.onChange}>
    </textarea>
  )
}

export default Input;