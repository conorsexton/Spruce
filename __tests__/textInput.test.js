import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from '../components/Input';

configure({ adapter: new Adapter() });

const onChangeMock = jest.fn();

describe('Text input', () => {
  let input;
  beforeAll(() => {
    input = shallow(<Input cols={70} rows={20} onChange={onChangeMock}/>)
  });

  it('Matches text input snapshot on initial render', () => {
    expect(input).toMatchSnapshot();
  });

  it('Renders a textarea element', () => {
    expect(input.type()).toBe('textarea');
  })

  it('Calls change handler on user input', () => {
    input.simulate('change');
    expect(onChangeMock).toHaveBeenCalled();
  })
});