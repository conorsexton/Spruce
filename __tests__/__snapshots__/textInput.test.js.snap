// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input Matches text input snapshot on initial render 1`] = `ShallowWrapper {}`;
